pkg/fleets/controller.go

```diff
@@ -0,0 +1,249 @@
+	gsSet := obj.(*stablev1alpha1.GameServerSet)
+	ref := metav1.GetControllerOf(gsSet)
+	if ref == nil {
+		return
+	}
+
+	fleet, err := c.fleetLister.Fleets(gsSet.ObjectMeta.Namespace).Get(ref.Name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			c.logger.WithField("ref", ref).Info("Owner Fleet no longer available for syncing")
+		} else {
+			runtime.HandleError(c.logger.WithField("fleet", fleet.ObjectMeta.Name).WithField("ref", ref),
+				errors.Wrap(err, "error retrieving GameServerSet owner"))
+		}
+		return
+	}
+	c.workerqueue.Enqueue(fleet)
+}
+
+// syncFleet synchronised the fleet CRDs and configures/updates
+// backing GameServerSets
+func (c *Controller) syncFleet(key string) error {
+	c.logger.WithField("key", key).Info("Synchronising")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		// don't return an error, as we don't want this retried
+		runtime.HandleError(c.logger.WithField("key", key), errors.Wrapf(err, "invalid resource key"))
+		return nil
+	}
+
+	fleet, err := c.fleetLister.Fleets(namespace).Get(name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			c.logger.WithField("key", key).Info("Fleet is no longer available for syncing")
+			return nil
+		}
+		return errors.Wrapf(err, "error retrieving fleet %s from namespace %s", name, namespace)
+	}
+
+	list, err := c.listGameServerSets(fleet)
+	if err != nil {
+		return err
+	}
+
+	var activeGsSet *stablev1alpha1.GameServerSet
+
+	// if there isn't a GameServerSet yet, then create one
+	if len(list) == 0 {
+		activeGsSet = fleet.GameServerSet()
+		activeGsSet, err = c.gameServerSetGetter.GameServerSets(fleet.ObjectMeta.Namespace).Create(activeGsSet)
+		if err != nil {
+			return errors.Wrapf(err, "error creating gameserverset for fleet %s", fleet.ObjectMeta.Name)
+		}
+
+		c.recorder.Eventf(fleet, corev1.EventTypeNormal, "CreatingGameServerSet",
+			"Created GameServerSet %s", activeGsSet.ObjectMeta.Name)
+
+	} else {
+		// for now, we're ignoring any change to the template - will handle on the next PR
+		// therefore, we are going to assume for the moment, that there is only ever one
+		// GameServerSet for a Fleet - we will handle multiple GameServerSets in the next PR
+		activeGsSet = list[0]
+	}
+
+	// if the replica count has changed, then update the GameServerSet
+	if fleet.Spec.Replicas != activeGsSet.Spec.Replicas {
+		gsSetCopy := activeGsSet.DeepCopy()
+		gsSetCopy.Spec.Replicas = fleet.Spec.Replicas
+
+		if gsSetCopy, err = c.gameServerSetGetter.GameServerSets(fleet.ObjectMeta.Namespace).Update(gsSetCopy); err != nil {
+			return errors.Wrapf(err, "error updating replicas for gameserverset for fleet %s", fleet.ObjectMeta.Name)
+		}
+		c.recorder.Eventf(fleet, corev1.EventTypeNormal, "ScalingGameServerSet",
+			"Scaling GameServerSet %s to %d", gsSetCopy.ObjectMeta.Name, gsSetCopy.Spec.Replicas)
```

Comment:
Could we add the old number to in the event so we know if it was scaling up or down?

Reply:
Nice! Done.